src/msg/async/ProtocolV2.cc

```diff
@@ -862,7 +862,8 @@ void ProtocolV2::prepare_send_message(uint64_t features,
   }
 
   // encode and copy out of *m
-  m->encode(features, messenger->crcflags);
+  // ProtocolV2 don't need crc of header.
```

Comment:
This part doesn't actually make sense to me — do we force-calculate it again? We don't want nonsense sitting in there even if we are protected by the crc mode at the protocol layer.

Reply:
@gregsfortytwo .  In protocolv2,  i use ceph_msg_header2  rather than ceph_msg_header. But in Message::encode, it calc crc of ceph_msg_header. Or am i missing something?

Reply:
I would target this fix at v1 only since @rzarzynski is rewriting the way crcs are handled for v2.